File: src/csharp/oop/FileSearch/Search/Operands/NotOperand.cs
using Microsoft.Extensions.FileProviders;

namespace FileSearch.Search.Operands;

public sealed class NotOperand(IMatchCriteria criteria) : OperandCriteriaBase
{
    public override bool IsMatch(IFileInfo fileInfo, SearchOptions options)
        => !criteria.IsMatch(fileInfo, options);
}

File: src/python/oop/FileSearch/search/operands/not_operand.py
from .operand_base import *
from ..predicate_base import *
from ..file_info import *
from ..search_options import *

class NotOperand(OperandBase):
    def __init__(self, criteria: PredicateBase):
        self.__criteria = criteria

    def is_match(self, file: FileInfo, options: SearchOptions = None) -> bool:
        return not self.__criteria(file)
